pkg/operation/botanist/component/logging/eventlogger/event_logger.go

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventlogger
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	gutil "github.com/gardener/gardener/pkg/utils/gardener"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	name                = "event-logger"
+	vpaName             = "event-logger-vpa"
+	managedResourceName = "event-logger"
+	roleName            = "gardener.cloud:logging:event-logger"
+
+	filter = `[FILTER]
+    Name                nest
+    Match               kubernetes.*` + v1beta1constants.DeploymentNameEventLogger + `*` + name + `*
+	Operation           lift
+	Nested_under        log
+[FILTER]
+	Name                record_modifier
+	Match               kubernetes.*` + v1beta1constants.DeploymentNameEventLogger + `*` + name + `*
+	Record 				job event-logging
+`
+)
+
+type eventLogger struct {
+	client         client.Client
+	namespace      string
+	image          string
+	replicas       int32
+	secretsManager secretsmanager.Interface
+}
+
+// NewEventLogger creates a new instance of event-logger for the shoot event logging.
+func NewEventLogger(
```

Comment:
```suggestion
func New(
```
As this is already in the `eventlogger` package.